### PR TITLE
Added login/logout

### PR DESCRIPTION
**Pull request by Motrunich**
Fix registration
added login/logout

### CODE REVIEW

File: templates/components/header.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <nav>
     <ul class="nav nav-pills float-right">
       <li class="nav-item">
-        <a class="nav-link" href="#">Личаківський</a>
```
Comment (aodarc) [inline templates/components/header.html:5]:
```
{% for district in districts %}
<li class="nav-item">
        <a class="nav-link" href="?name={{district.name}}">{{district.name}}</a>
 </li>
{%endfor%}
```

File: apps/kindergarten/views.py [hunk 2]
```diff
@@ -1,11 +1,5 @@
...
-from apps.kindergarten.models import Kindergarten
-
-
-class Kindergartenview(ListView):
-    model= Kindergarten
-    template_name = 'kindergarden/kindergarten_page.html'
-    context_object_name = 'kinder_list'
-    queryset = Kindergarten.objects.select_related('district').all()
\ No newline at end of file
+# class KindergartenView(ListView):
```
Comment (aodarc) [inline apps/kindergarten/views.py:1]: uncomment

File: apps/user_profile/views.py [hunk 3]
```diff
@@ -1,14 +1,40 @@
...
-class RegisterFormView(FormView):
-    form_class = UserCreationForm
-    template_name = "main_page/register.html"
-    success_url = reverse_lazy('main')
 
-    def form_valid(self, form):
-        form.save()
-        return super(RegisterFormView, self).form_valid(form)
+@login_required
+def home(request):
```
Comment (aodarc) [inline apps/user_profile/views.py:12]: remove it

File: apps/user_profile/views.py [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
+            raw_password = form.cleaned_data.get('password1')
+            user = authenticate(username=username, password=raw_password)
+            login(request, user)
+            return redirect('main')
+    else:
+        form = SignUpForm()
+    return render(request, 'main_page/signup.html', {'form': form})
+
+
+class DistrictView(ListView):
```
Comment (aodarc) [inline apps/user_profile/views.py:31]: do we need this?

File: apps/user_profile/views.py [hunk 5]
```diff
@@ -1,14 +1,40 @@
...
+    model = Kindergarten
+    template_name = 'main_page/distrOne.html'
+    context_object_name = 'kinder_list'
+
+
+def search_kinder(request):
```
Comment (aodarc) [inline apps/user_profile/views.py:37]: move this logic to MainPageView

File: final_project/settings.py [hunk 6]
```diff
@@ -62,6 +63,7 @@
                 'django.template.context_processors.request',
                 'django.contrib.auth.context_processors.auth',
                 'django.contrib.messages.context_processors.messages',
+                # 'django.core.context_processors.media',
```
Comment (aodarc) [inline final_project/settings.py:66]: clear it

File: final_project/settings.py [hunk 7]
```diff
@@ -117,9 +119,14 @@
...
 STATIC_URL = '/static/'
 STATIC_ROOT = 'dist'
 
 STATICFILES_DIRS = (
     os.path.join(BASE_DIR, "project_static"),
-    )
\ No newline at end of file
+    )
+
+EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
```
Comment (aodarc) [inline final_project/settings.py:130]: it is default value

File: final_project/settings.py [hunk 8]
```diff
@@ -117,9 +119,14 @@
...
+
+LOGIN_REDIRECT_URL = 'main'
```
Comment (aodarc) [inline final_project/settings.py:132]: LOGIN_REDIRECT_URL = reverse_lazy('main')

File: templates/components/header.html [hunk 9]
```diff
@@ -2,7 +2,7 @@
...
+          <a class="nav-link" href="{% url 'search' %}">Личаківський</a>
```
Comment (aodarc) [inline templates/components/header.html:5]: use ?name=

File: templates/main_page/distrOne.html [hunk 10]
```diff
@@ -0,0 +1,49 @@
+{% extends 'base.html' %}
```
Comment (aodarc) [inline templates/main_page/distrOne.html:1]: move this search form to main page

**Review by aodarc — CHANGES_REQUESTED** [review]
check comments

File: apps/user_profile/views.py (hunk 3 above)
Comment (Motrunich) [reply apps/user_profile/views.py:12]: Done

File: final_project/settings.py (hunk 6 above)
Comment (Motrunich) [reply final_project/settings.py:66]: Done

File: final_project/settings.py (hunk 7 above)
Comment (Motrunich) [reply final_project/settings.py:130]: Done. Deleted

File: apps/kindergarten/views.py (hunk 2 above)
Comment (Motrunich) [reply apps/kindergarten/views.py:1]: Done

File: apps/user_profile/views.py (hunk 5 above)
Comment (Motrunich) [reply apps/user_profile/views.py:37]: Done

File: templates/components/header.html (hunk 1 above)
Comment (Motrunich) [reply templates/components/header.html:5]: Done

File: apps/user_profile/views.py (hunk 4 above)
Comment (Motrunich) [reply apps/user_profile/views.py:31]: deleted

File: templates/components/header.html (hunk 9 above)
Comment (Motrunich) [reply templates/components/header.html:5]: done

File: templates/main_page/distrOne.html (hunk 10 above)
Comment (Motrunich) [reply templates/main_page/distrOne.html:1]: done

File: final_project/settings.py (hunk 8 above)
Comment (Motrunich) [reply final_project/settings.py:132]: done

**Review by maziar0andriy — APPROVED** [review]

**Pull request closed without merging**